### ISSUE TITLE
Logging in geotrellis.raster

### ISSUE THREAD

**Issue by lossyrob** [geotrellis/geotrellis#1408]
We currently don't have a good way to log in geotrellis.raster. There are fears that our logging dependency, scala-logging-slf4j, will have serialization issues with spark.

One thing we need to log is when a UInt32 geotiff is read in, and converted to a floating point tile; the user should be warned.

**Comment by echeipesh** [comment]
The trick I have used is to make a logger trait that uses `@transient lazy val` for the logger:

```scala
import com.typesafe.scalalogging._
import com.typesafe.scalalogging.slf4j._
import org.slf4j.LoggerFactory

trait Logging {
  @transient lazy val logger = Logger(LoggerFactory.getLogger(getClass.getName))
}
```

This essentially means that the logger will not be serialized and will be re-created when used on the other side of the serialization.

**Comment by moradology** [comment]
That's the trick I've seen people using in older versions of `scala-logging`/`sl4j`/etc - it is worth looking into newer versions of our logging libraries because (at least in scala-logging 3.4.0) some of them provide nicely serializable loggers

**Comment by echeipesh** [comment]
Unfortunately `scala-logging` 3.4 is not published for 2.10. They stopped doing that sometime in 2.* line. https://github.com/typesafehub/scala-logging#prerequisites
